hibernate-reactive-core/src/main/java/org/hibernate/reactive/pool/impl/SqlClientConnection.java

```diff
@@ -175,10 +175,19 @@
 			int i = 0;
 			RowSet<Row> resultNext = result;
 			if ( parametersBatch.size() > 0 ) {
-				do {
-					updateCounts[ i++ ] = resultNext.rowCount();
-					resultNext = resultNext.next();
-				} while ( resultNext != null && i < parametersBatch.size() );
+				final RowIterator<Row> iterator = resultNext.iterator();
+				if ( iterator.hasNext() ) {
+					while ( iterator.hasNext() ) {
+						updateCounts[i++] = iterator.next().getInteger( 0 );
+					}
+					resultNext = null;
+				}
+				else {
+					do {
+						updateCounts[i++] = result.rowCount();
+						resultNext = resultNext.next();
+					} while ( resultNext != null && i < parametersBatch.size() );
+				}
```

Comment:
@Blafond I cannot remember why we need to distinguish between the two cases, could you add some comments, please?

Reply:
@DavideD  For the *result* object using Oracle, the *next* value is NULL but the *list* is populated.
```
result = {RowSetImpl@8350} 
 list = {ArrayList@8354}  size = 20
 updated = 20
 columnNames = {Collections$EmptyList@8355}  size = 0
 columnDescriptors = {Collections$EmptyList@8355}  size = 0
 size = 20
 next = null
 properties = null
```
The fix checks *hasNext()* first and fills the updateCounts[] array

Reply:
Ok, but what's the test about? When does this happen?

Reply:
Most of the tests in `BatchingConnectionTest.java` are failing, for instance

Reply:
```
        Caused by:
        java.lang.IllegalStateException: HR000059: Number of results is greater than number of batched parameters
            at org.hibernate.reactive.pool.impl.SqlClientConnection.lambda$updateBatch$4(SqlClientConnection.java:198)
            at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:642)
```